app/assets/javascripts/admin/addon/components/site-setting.gjs

```diff
@@ -124,7 +124,18 @@ export default class SiteSettingComponent extends Component {
   }
 
   get overridden() {
-    return this.setting.default !== this.buffered.get("value");
+    if (
+      this.setting.json_schema ||
+      this.setting.schema ||
+      this.setting.objects_schema
+    ) {
+      return !deepEqual(this.setting.default, this.buffered.get("value"));
+    } else {
+      return (
+        this.setting.default?.toString() !==
+        this.buffered.get("value")?.toString()
+      );
+    }
```

Comment:
There is some logic duplication here and in the `dirty` getter. Perhaps we should extract the logic into a `#isValueDirty`? Something like the following:

```
#isValueDirty(oldValue, newValue) {
  if (
    this.setting.json_schema ||
    this.setting.schema ||
    this.setting.objects_schema
  ) {
    return !deepEqual(oldValue, newValue);
  } else {
    return (
      oldValue?.toString() !== newValue?.toString()
    );
  } 
}
```

Reply:
this is done.